### DISCUSSION TITLE
SAP dropped support for OrientDB on Sept 1 2021

### DISCUSSION THREAD

**Discussion by n0mer** [orientechnologies/orientdb#9735] [General]
Quote from https://en.wikipedia.org/wiki/OrientDB
 
> On September 15, 2017, OrientDB LTD company has been acquired by CallidusCloud a public company traded on NASDAQ.

> On January 30, 2018, it was announced SAP (company) acquired CallidusCloud for $2.4 billion. and therefore OrientDB is now supported by SAP (company).
 

> On September 1st, 2021, the original founder Luca Garulli left SAP (company) and forked the project into ArcadeDB after **SAP decided to stop providing commercial support for OrientDB**.
 

https://blog.arcadedb.com/welcome-to-arcadedb  new project home

**Comment by andrii0lomakin** [comment]
ArcadeDB is a separate project which shares the same model. It is not a fork of OrientDB. Would be strange to call MySQL to be a fork of PostreSQL only because they share the same model.

**Comment by andrii0lomakin** [comment]
Also as you can see ODB team is still here and still working on the project. And it is quite past of Sept 1 2021.

**Comment by lvca** [comment]
The page on Wikipedia reported "SAP decided to stop providing commercial support for OrientDB." - And this is 100% factually true. SAP doesn't provide commercial support for OrientDB anymore. Try contact yourself SAP and ask for it :-) Or @laa please provide the link where I can buy support.

There is no shame in telling the truth.

Perhaps everybody can see only 3 people (and 1 bot) are contributing to the code base: https://github.com/orientechnologies/orientdb/pulse/monthly.

**Comment by andrii0lomakin** [comment]
@lvca truth is that ArcadeDB is not a fork of OrientDB. And truth is that SAP still pays for the team to provide support of open source project and if we check ArcadeDB GitHub page we see that it has only a single contributor and bot itself. My main message that ArcadeDB is not fork of OrientDB it is a completely separate project with single major contributor.

**Comment by lvca** [comment]
@laa ArcadeDB was born as a fork of OrientDB: same idea, similar features but only took the code that was worth saving. There is no definition about how much of the original codebase must be used to be considered a "fork". ArcadeDB codebase today is only about 10% the same as OrientDB, mainly only the SQL engine is the same with some small adaptations. This means all the applications developed by using SQL (and/or Gremlin) are >90% portable and that was the goal.

About OrientDB, it's nice that SAP still continues to pay a few of you to support the existing clients, but as I said, **there is no support for new clients**, the product was removed from the official price list a long time ago. There is no trace of OrientDB on the SAP website and https://orientdb.com/ is still a static ugly page that for more than 2 years has not been updated.

I think OrientDB users deserve better than this. 

An official press release about the official position of SAP would be the best for OrientDB users, but I think SAP is too busy with its own business that has no time to do a proper one.

If an OrientDB user is looking for a community with a future, that supports the product by fixing bugs quickly and introducing new features following the community wishes, ArcadeDB is the right place.

**Comment by n0mer** [comment]
@laa i sent e-mail to orientdb_enquiries@sap.com on "Wed 01-Dec-21" - no reply for 2 weeks. Will re-post question here, maybe this place is more visible:
 

I read here https://groups.google.com/g/orient-database/c/fmo0WKnfXUc message from orientdb contributor "Luigi Dell'Aquila" that v2 is EOL’ed.

1. Is that true?

2. If so - what are the steps to migrate from 2.x to 3.x?

3. What amount of support does your enterprise subscription practically offers in such upgrade?

**Comment by andrii0lomakin** [comment]
@lvca  If one database uses only SQL code of another database that is not a fork :-) that is a completely new database with its own principles and only one library borrowed.  Only you alone push the code into the repository, which is clearly seen from the commit list. No one from current and former developers agreed to participate in the development of ArcadeDB.

@n0mer as for your questions:
1. Yes v2 is EOL.
2. Ideally you need to export data to and from JSON in related versions. 
3. If you have an EE subscription it is fully supported by now, but you can not buy a new subscription.

**Reply by n0mer** [reply]
@laa Andrey, can you please provide link to v2 EOL announcement? If that came via e-mail - please forward it.

Looking for any guidance on how to keep using OrientDB having said v2 is no longer maintained.

**Comment by andrii0lomakin** [comment]
@n0mer do you have any other questions?

**Reply by n0mer** [reply]
Let's continue over e-mail - i still expect some response, as well as other colleagues Cc:-ed

**Reply by lvca** [reply]
Let us know how it goes 😆

**Reply by andrii0lomakin** [reply]
Hi @n0mer I do not see your email. Could you send it to me again? lomakin.andrey@gmail.com

**Reply by andrii0lomakin** [reply]
Or better a.lomakin@sap.com . It will be easier for me to handle it.

**Comment by smolinari** [comment]
I'm confused and sad to get wind of all this. 

I'm confused, because it sounds like SAP is letting OrientDB "sunset", despite the valiant efforts of the couple three devs here (who I have great respect for). What can be said, to counter this impression? Is it incorrect?

I'm sad, because it seems a slight animosity between you Luca, and the team that stayed by your side for so long. Maybe that impression is also incorrect? It seems and makes me also sad, to see you are also a bit frustrated from the situation overall i.e. having to break up with your baby. I guess there is a whole story behind the breakup? At any rate, I wish you all the best. 

Scott

**Reply by lvca** [reply]
Hey @smolinari I don't feel comfortable revealing details about how OrientDB was "absorbed" by SAP and the last 4 years on this journey... And actually, they don't really matter right now. What matters is to give continuity to the previous OrientDB users and to welcome brand new users to the Multi-Model world with a new product that is much faster, more stable, and easy to maintain than OrientDB. Also, I'm experimenting with running ArcadeDB as a foundation where the incoming from the paying sponsors are shared between contributors to have a sustainable Open Source model hoping that it could be adopted by other open-source projects as well: "Less corporations and more individuals in the Open Source".

**Reply by smolinari** [reply]
@lvca - Nice. And for sure, this isn't a place for M&A details. I'm glad you are happy working on a newer better graph database. I'm giving ArcadeDB a look as we speak. 😁  

Scott

**Comment by lvca** [comment]
Unbelievable how hard it was to have a public statement about the future of OrientDB by SAP... 

Finally, today we have something: "With SAP maintenance ending on 31.12.2023 existing subscription contracts will not be renewed.". Check the article https://orientdb.org/enterprise-agent-open-source.

**Reply by smolinari** [reply]
Does that mean OSS users can get Enterprise features now too? Does the "maintenance" ending mean SAP is no longer also going to pay for dev time to improve the database?

Scott

**Reply by andrii0lomakin** [reply]
Hi,
Does that mean OSS users can get Enterprise features now too? - yes.
As for the second question you need to ask SAP officials. The only thing which I can confirm is that at least till 31.12.2023 EE database will be maintained as usual.

**Reply by mdre** [reply]
Basically it is not maintained. We have a paid support and we have reported problems last year and there is still no solution. Even when one of that errors produces the database shutdown with an exception.
Just a simple Lucene query drop the engine.
